modules/ding_nodelist/ding_nodelist.module

```diff
@@ -0,0 +1,2477 @@
+  // Add new node to list.
+  $form_state['values']['selected_nodes_count'] = count($form_state['values']['selected_nodes']) + 1;
+  unset($form_state['input']['selected_nodes']);
+  foreach ($form_state['values']['selected_nodes'] as $key => $node) {
+    if (!empty($node['node'])) {
+      $tmp_nid = explode(':', $node['node']);
+      $tmp_nid = array_pop($tmp_nid);
+      $tmp_nid = substr($tmp_nid, 0, -1);
+      // TODO: Implement more complex check for given title + ID.
+      if (ctype_digit($tmp_nid)) {
+        $form_state['values']['selected_nodes'][$key]['nid'] = $tmp_nid;
+      }
+    }
+  }
+
+  // Copy 'input' since limit_validation_errors empties 'values'.
+  $form_state['values'] += $form_state['input'];
+  _ding_nodelist_content_type_form_save($form_state, TRUE);
+}
+
+/**
+ * Custom submit function for adding more filters.
+ */
+function ding_nodelist_add_filter_submit($form, &$form_state) {
+  // Add new filter.
+  $form_state['values']['taxonomy_filters_count'] = count($form_state['values']['taxonomy_filters']) + 1;
+
+  // Copy 'input' since limit_validation_errors empties 'values'.
+  $form_state['values'] += $form_state['input'];
+  _ding_nodelist_content_type_form_save($form_state, TRUE);
+}
+
+/**
+ * Custom submit function for filter deletion.
+ */
+function ding_nodelist_filter_delete_submit($form, &$form_state) {
+
+  $deleted_filter = $form_state['triggering_element']['#parents'][1];
+  unset($form_state['values']['taxonomy_filters'][$deleted_filter]);
+  unset($form_state['input']['taxonomy_filters'][$deleted_filter]);
+
+  // Re-index after deletion and save.
+  $form_state['input']['taxonomy_filters'] = array_values($form_state['input']['taxonomy_filters']);
+  $form_state['values'] = $form_state['input'];
+  _ding_nodelist_content_type_form_save($form_state, TRUE);
+}
+
+/**
+ * Custom submit function for filter deletion.
+ */
+function ding_nodelist_node_delete_submit($form, &$form_state) {
+  $deleted_node = $form_state['triggering_element']['#parents'][1];
+  unset($form_state['values']['selected_nodes'][$deleted_node]);
+  unset($form_state['input']['selected_nodes'][$deleted_node]);
+
+  // Re-index after deletion and save.
+  $form_state['input']['selected_nodes'] = array_values($form_state['input']['selected_nodes']);
+  $form_state['values'] = $form_state['input'];
+  _ding_nodelist_content_type_form_save($form_state, TRUE);
+}
+
+/**
+ * Custom submit function for adding more links.
+ */
+function ding_nodelist_add_bottom_link_submit(&$form, &$form_state) {
+  // Add new link to list.
+  $form_state['values']['bottom_links_count'] = count($form_state['values']['more_links']) + 1;
+  unset($form_state['input']['more_links']);
+
+  // Saving all the values of the form.
+  _ding_nodelist_content_type_form_save($form_state, TRUE);
+}
+
+/**
+ * Edit form validation function.
+ */
+function ding_nodelist_content_type_edit_form_validate($form, &$form_state) {
+  $form_state['restore'] = TRUE;
```

Comment:
This seems linked to previous weirdness. At least it should be documented why it is necessary to set this value here.